pkg/oauthserver/oauth/external/gitlab/gitlab.go

```diff
@@ -133,3 +119,49 @@ func (p *provider) GetUserIdentity(data *osincli.AccessData) (authapi.UserIdenti
 
 	return identity, true, nil
 }
+
+func (p *provider) getUserData(token string) (*gitlabUser, error) {
+	// try the v4 API first
+	userdata, err := p.getUserDataURL(token, p.userAPIURLV4)
+	if err == nil {
+		return userdata, nil
+	}
```

Comment:
See above, plus we shouldn't fall back for every error, but just for errors that indicate the endpoiont does not exist.
If it does and it returns an error we should just return the error and not try any fallback.

Reply:
> just for errors that indicate the endpoint does not exist

A 404?